src/transactions/feed/TransferFeedIcon.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import { useSelector } from 'react-redux'
+import ContactCircle from 'src/components/ContactCircle'
+import CoinbasePayIcon from 'src/icons/CoinbasePayIcon'
+import { coinbasePaySendersSelector } from 'src/recipients/reducer'
+import { FeedTokenTransfer } from 'src/transactions/feed/TransferFeedItem'
+import { useTransactionRecipient } from 'src/transactions/transferFeedUtils'
+import { TokenTransactionTypeV2 } from 'src/transactions/types'
+
+const AVATAR_SIZE = 40
+
+interface Props {
+  transfer: FeedTokenTransfer
+}
+
+export default function TransferFeedIcon({ transfer }: Props) {
+  const { type, address } = transfer
+
+  const coinbasePaySenders = useSelector(coinbasePaySendersSelector)
+  const recipient = useTransactionRecipient(transfer)
+
+  switch (type) {
+    // Checks if feed item is a received transaction from Coinbase
+    case TokenTransactionTypeV2.Received: {
+      const isCoinbasePaySender = coinbasePaySenders.includes(address)
+
+      if (isCoinbasePaySender) {
```

Comment:
if this evaluates to false, I think the function will return undefined

Reply:
I imitated the switch case statement from [here](https://github.com/valora-inc/wallet/blob/main/src/transactions/TransferFeedIcon.tsx). I believe if the if statement evaluates to false, default will be called.